src/components/user/AccountTabs.vue

```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import BaseButton from "@/components/global/BaseButton.vue";
+
+const props = withDefaults(
+  defineProps<{
+    activeTab: string;
+  }>(),
+  {
+    activeTab: "personalDetails",
+  }
+);
+
+const emit = defineEmits<{
+  (e: "setActiveTab", tab: string);
+}>();
+
+const changeTab = (tab: string) => {
+  emit("setActiveTab", tab);
+};
+</script>
+
+<template>
+  <section class="account_tabs">
+    <BaseButton
+      class="account__tab"
+      :class="{ active: activeTab === 'personalDetails' }"
+      @click="changeTab('personalDetails')"
+      >Personal Details</BaseButton
+    >
+    <BaseButton
+      class="account__tab"
+      :class="{ active: activeTab === 'reservations' }"
+      @click="changeTab('reservations')"
+      >Reservations</BaseButton
+    >
+  </section>
+</template>
+
+<style lang="scss" scoped>
+.account_tabs {
```

Comment:
We should stick to kebab-case in our normal classes 😄 

```suggestion
.account-tabs {
```